packages/core/src/caches/base-cache.ts

```diff
@@ -51,7 +51,10 @@ export abstract class BaseCache<CacheMapT extends Record<string, unknown>> {
   ): Promise<Optional<CacheMapT[Type]>> {
     return trySafe(async () => {
       const data = await this.cacheStore.get(this.cacheKey(type, key));
-      return this.getValueGuard(type).parse(JSON.parse(data ?? ''));
+      // Allow `null` value to be stored in the cache.
+      // If the value is `null`, we should return `null` instead of parsing it as JSON.
+      // Otherwise, treat the value as JSON and parse it.
+      return this.getValueGuard(type).parse(data === 'null' ? null : JSON.parse(data ?? ''));
```

Comment:
<img width="152" alt="image" src="https://github.com/user-attachments/assets/d831fd31-c3cc-4994-9af1-b754cb15c55f" />

do we actually need this ternary?

Reply:
updated